src/Build.UnitTests/BackEnd/SdkResolverService_Tests.cs

```diff
@@ -539,29 +603,82 @@ public void IsRunningInVisualStudioIsSetForResolverContext()
 
         private class MockLoaderStrategy : SdkResolverLoader
         {
-            private readonly bool _includeErrorResolver;
+            private List<SdkResolver> _resolvers;
+            private List<Tuple<string, SdkResolver>> _resolversWithPatterns;
```

Comment:
nit: create modern tuples with element names so you don't have to do the ugly `Item1`/`Item2` stuff below

```suggestion
            private List<(string NamePattern, SdkResolver Resolver)> _resolversWithPatterns;
```